go/ai/request_helpers.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ai
+
+func NewUserTextGenerateRequest(text string) *GenerateRequest {
+	return &GenerateRequest{
+		Messages: []*Message{
+			{
+				Role:    RoleUser,
+				Content: []*Part{NewTextPart(text)},
+			},
+		},
+	}
+}
+
+func NewGenerateRequest(config map[string]any, messages ...*Message) *GenerateRequest {
+	return &GenerateRequest{
+		Config:   config,
+		Messages: messages,
+	}
+}
+
+func NewUserMessage(parts ...*Part) *Message {
```

Comment:
I don't think most of these variations are going to be that useful. `NewMessage` seems like enough.
Maybe also `NewUserMessage` if that's very common.
The rest seem too rare to be worth the extra API.

Reply:
The user code flows much better when it's readable up front. Since there are only 3 role types, I think these helpers are useful for reducing friction.

Reply:
`NewUserMessage` and `NewSystemMessage` are very common so I'll argues that we definitely need those. `NewModelMessage` is less useful, but if we have the other two I don't see why not have it as well. We'll have to revisit Tools messages....
I'm not a fan of using `NewMessage` because having to specify the role looks very noisy: `ai.NewMessage(ai.RoleUser, ...)`. I think it's work making sure there are fewer `ai.`s....